velox/functions/lib/Re2Functions.cpp

```diff
@@ -381,6 +406,235 @@ class Re2SearchAndExtract final : public VectorFunction {
   const bool emptyNoMatch_;
 };
 
+class Re2ReplaceConstantPattern final : public VectorFunction {
+ public:
+  /**
+   * @brief Constructor for the Re2ReplaceConstantPattern class.
+   *
+   * This class provides a method to replace occurrences of a pattern in a
+   * string starting from a given position.
+   *
+   * @param basePattern The regular expression pattern to search for in the
+   * input string.
+   */
+  explicit Re2ReplaceConstantPattern(StringView basePattern)
+      : basePattern_(basePattern.getString()) {}
+
+  /**
+   * @brief Applies the regex replace operation on the input string vectors.
+   *
+   * This function reads the input strings and replaces occurrences of the regex
+   * pattern starting from the given position. If the position is not provided,
+   * it defaults to 0. The result of the replacement operation is written to the
+   * resultRef vector.
+   *
+   * @param rows The rows of data to be processed.
+   * @param args The arguments passed to the function.
+   *             args[0]: The input strings.
+   *             args[1]: The pattern to search for.
+   *             args[2]: The replacement strings.
+   *             args[3]: (Optional) The starting position.
+   * @param context The evaluation context.
+   * @param resultRef The vector to write the result to.
+   */
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /*outputType*/,
+      EvalCtx& context,
+      VectorPtr& resultRef) const final {
+    VELOX_CHECK(args.size() == 3 || args.size() == 4);
+
+    // Decode the input vector containing the strings on which the replace
+    // operation is to be performed.
+    exec::LocalDecodedVector input(context, *args[0], rows);
+    // Ensure a writable output vector of type Array(VARCHAR) is prepared to
+    // receive the results.
+    BaseVector::ensureWritable(
+        rows, ARRAY(VARCHAR()), context.pool(), resultRef);
+    exec::VectorWriter<Array<Varchar>> resultWriter;
+    resultWriter.init(*resultRef->as<ArrayVector>());
+
+    // Decode the overwrite vector containing the strings to replace matched
+    // patterns with.
+    exec::LocalDecodedVector overwriteVec(context, *args[2], rows);
+    std::vector<re2::StringPiece> overwrite = {
+        toStringPiece(overwriteVec->valueAt<StringView>(0))};
+
+    // Default position is 1 unless specified.
+    int64_t position = 1;
+    if (args.size() == 4) {
+      // Decode the position vector containing the starting position for the
+      // regex replacement.
+      exec::LocalDecodedVector positionVec(context, *args[3], rows);
+      position = positionVec->valueAt<int64_t>(0);
```

Comment:
Seems we can't assume position is `int64_t` as we also support `INTEGER` in signature. Right?

Reply:
Removed INTEGER support. BIGINT covers.

As for moving functions to Spark, it seems like all Spark regex functions are located in velox/functions/lib. How should we handle this?

Reply:
We don't need a copy for Spark if there is NO semantics mismatch between Spark and Presto.

Reply:
Moved implementation to just Spark as PrestoSQL allows for a function to be passed as the replacement argument which is not available for Spark